net/dns/resolver/forwarder.go

```diff
@@ -200,18 +202,45 @@ type forwarder struct {
 	// /etc/resolv.conf is missing/corrupt, and the peerapi ExitDNS stub
 	// resolver lookup.
 	cloudHostFallback []resolverAndDelay
+
+	// Control D specific fields. These allow the admin to destinguish between
+	// different clients using the same DoH resolver URL
+	dohClientHostname string // Hostname to send in Control D DoH requests
+	dohClientMac      string // MAC address to send in Control D DoH requests
+	dohClientIP       string // IP address to send in Control D DoH requests
 }
 
 func init() {
 	rand.Seed(time.Now().UnixNano())
 }
 
 func newForwarder(logf logger.Logf, netMon *netmon.Monitor, linkSel ForwardLinkSelector, dialer *tsdial.Dialer) *forwarder {
+	hostname, _ := os.Hostname()
+	tailscaleIP := ""
+	clientMac := ""
+
+	// TODO: is this the best way to determine the tailscale IP of this node?
+	if tsIPs, _, err := interfaces.Tailscale(); err == nil && len(tsIPs) > 0 {
+		tailscaleIP = tsIPs[0].String()
+	}
+
+	// TODO: is this the best way to determine the MAC address of this node?
+	if idx, err := interfaces.DefaultRouteInterfaceIndex(); err == nil {
+		iface, err := net.InterfaceByIndex(idx)
+		if err == nil {
+			clientMac = iface.HardwareAddr.String()
+		}
+	}
```

Comment:
Remove all this. We'd push these parameters to the node from the control plane, like we do with NextDNS.

Reply:
Can you please link me to where this control plane is doing this for NextDNS?

Also, we currently only support passing full metadata (local IP, MAC and hostname) about the client via DoH headers, but we do support adding just the hostname via the DoH URI, i.e:

https://dns.controld.com/<resolver_id>/<hostname>

Would DoH headers with full metadata be completely out of the question? If so, would love to lear n more about what we can do with the control plane.